akka-actor/src/main/scala/akka/routing/Resizer.scala

```diff
@@ -71,6 +94,7 @@ case object DefaultResizer {
       backoffRate = resizerConfig.getDouble("backoff-rate"),
       messagesPerResize = resizerConfig.getInt("messages-per-resize"))
 
+  @deprecated("User Resizer.fromConfig instead", since = "2.4.1")
```

Comment:
I wasn't sure here as well. I made this deprecation mostly due to API design. 
I could make `DefaultResizer.fromConfig` to return either a `DefaultConfig` or a `MetricsBasedResizer` based on configuration, but I didn't like this API design. Moving this `fromConfig` method to `Resizer` provides an API that tells people there are multiple implementations of the `Resizer` trait.
It appeared to me that this method is mostly used internally, and that this deprecation would have low short-term impact. But if that's not the case, I can remove this deprecation and make it only returns a `DefaultResizer`, the `Resizer.fromConfig` can continue return either `MetricsBasedResizer` or `DefaultResizer` based on configuration.